### PR TITLE
Add soft kill option to all processes

### PR DESCRIPTION
**Pull request by yabola**
### What changes are proposed in this pull request?

Before:
The behavior of the stop process scripts is: When the execution (**using kill pid**) exceeds the maxWait duration , the process will be forced to kill (**kill -9 pid**).
After this RP: 
We can add option param '-s' . When the execution exceeds the maxWait duration, the process won't be forcibly killed.
for example:
`sh bin/alluxio-stop.sh worker -s`
`sh bin/alluxio-stop.sh all -s`

### Why are the changes needed?
Make the kill command more gracefully.
Parent PR and context: https://github.com/Alluxio/alluxio/pull/13373.
related PR: https://github.com/Alluxio/alluxio/pull/13876
We need to umount fuse properly instead of kill -9 pid 

### Does this PR introduce any user facing changes?
Yes. Add one option for` alluxio-stop.sh`
Usage: alluxio-stop.sh  [component] [-c cache] [-s]

### CODE REVIEW

**Comment by yabola** [comment]
My test step(I add test log for see what happened)
`sh bin/alluxio-stop.sh all -s` -->
image

My umount fuse test step( after merge https://github.com/Alluxio/alluxio/pull/13876)
1. embed mounted dir:
```
alluxio.worker.fuse.enabled=true
alluxio.worker.fuse.mount.alluxio.path=/fuse_data
alluxio.worker.fuse.mount.point=/Users/chenliang.lu/soft/alluxio_data/fuse_data
```
2. open file on fuse dir:
`less /Users/chenliang.lu/soft/alluxio_data/fuse_data/a.txt`
3. stop worker with `-s` option:
   `sh bin/alluxio-stop.sh worker -s` -> worker process doesn't be killed.
image

File: bin/alluxio [hunk 1]
```diff
@@ -40,7 +40,7 @@ function printUsage {
   echo -e "                \t NOTE: This command requires a valid ufs path to run tests against."
   echo -e "  readJournal   \t Read an Alluxio journal file from stdin and write a human-readable version of it to stdout."
   echo -e "  upgradeJournal\t Upgrade an Alluxio journal version 0 (Alluxio version < 1.5.0) to an Alluxio journal version 1 (Alluxio version >= 1.5.0)."
-  echo -e "  killAll <WORD>\t Kill processes containing the WORD."
+  echo -e "  killAll [-s] <WORD>\t Kill processes containing the WORD (if '-s' specified, processes won't be forcibly killed even if operation is timeout)."
```
Comment (LuQQiu) [inline bin/alluxio:43]: processes that timeout with the termination signal won't be killed.

**Review by ZhuTopher — APPROVED** [review]
LGTM, thanks for the improvement!

**Review by ZacBlanco — CHANGES_REQUESTED** [review]

File: bin/alluxio [hunk 2]
```diff
@@ -79,31 +79,52 @@ EOF
 }
 
 function killAll {
-  if [[ $# -ne 1 ]]; then
-    echo "Usage: alluxio killAll <WORD>" >&2
+  if [[ $# -gt 2  ]]; then
+    echo "Usage: alluxio killAll [-s] <WORD> (if '-s' specified, processes won't be forcibly killed even if operation is timeout.)" >&2
```
[the remark this answers is not in the document]
Comment (yabola) [reply bin/alluxio]: thanks, down

**Review by ZacBlanco — APPROVED** [review]

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by LuQQiu** [comment]
Does any user-facing document need to be added/updated?
Where do we suggest users use this soft kill option? or it's only designed for advanced users?

**Comment by ZhuTopher** [comment]
@LuQQiu I don't believe we have any user-facing docs which detail how to use these scripts; it's just the `USAGE:` message in the scripts themselves. If there is a direct use-case for which soft-kill would be useful then we should mention it in the corresponding use-case's doc (if it exists), but otherwise idk if there's anything more to be added. wdyt?

**Pull request merged**